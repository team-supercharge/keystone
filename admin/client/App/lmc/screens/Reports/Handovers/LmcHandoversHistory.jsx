import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class LmcHandoversHistory extends Component {
    render () {
        const { handovers } = this.props
        return (
            <div>
                { JSON.stringify(handovers) }
            </div>
        )
    }
}

LmcHandoversHistory.propTypes = {
    handovers: PropTypes.array.isRequired
}