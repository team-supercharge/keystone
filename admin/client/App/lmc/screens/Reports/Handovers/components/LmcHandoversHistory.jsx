import React, { Component } from 'react'
import PropTypes from 'prop-types'
import LmcHandoverHistoryItem from './LmcHandoverHistoryItem.jsx'

export default class LmcHandoversHistory extends Component {
    render () {
        const { handovers } = this.props
        return (
            <div>
                { Object.keys(handovers).map((date, i) => {
                    const currentHandoversList = handovers[date]

                    return (
                        <div key={i}>
                            <h2 style={styles.heading}>
                                {date}
                            </h2>
                            <div 
                                className='lmc-theme-gradient' 
                                style={styles.divider} 
                            />
                            { currentHandoversList.map((handover, i) => {
                                return (
                                    <LmcHandoverHistoryItem
                                        handover={handover}
                                        key={i}
                                    />
                                )
                            })}
                        </div>
                    )
                })}
            </div>
        )
    }
}

const styles = {
    divider: {
        height: 2,
        marginBottom: 22,
        width: '100%',
    },
    heading: {
        marginBottom: '0.3em',
        fontWeight: 300,
        textOverflow: 'ellipsis',
        hyphens: 'auto',
    },
}

LmcHandoversHistory.propTypes = {
    handovers: PropTypes.array.isRequired
}