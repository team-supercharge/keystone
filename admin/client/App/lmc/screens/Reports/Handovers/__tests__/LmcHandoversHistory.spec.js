import React from 'react'
import { shallow } from 'enzyme'
import LmcHandoversHistory from '../components/LmcHandoversHistory.jsx'

describe('LmcHandoversHistory', () => {
    let wrapper
    let handovers

    beforeEach(() => {
        handovers = [
            { id: 'TestId', createdBy: 'TestCarer', witnessedBy: 'TestCarer2' }
        ]
        wrapper = shallow(
            <LmcHandoversHistory
                handovers={handovers}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})