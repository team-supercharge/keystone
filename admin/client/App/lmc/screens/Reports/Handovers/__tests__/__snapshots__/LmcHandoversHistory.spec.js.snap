// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcHandoversHistory renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div>
      <h2
        style={
          Object {
            "fontWeight": 300,
            "hyphens": "auto",
            "marginBottom": "0.3em",
            "marginTop": 20,
            "textOverflow": "ellipsis",
          }
        }
      >
        Mar 11th
      </h2>
      <div
        className="lmc-theme-gradient"
        style={
          Object {
            "height": 2,
            "marginBottom": 12,
            "width": "100%",
          }
        }
      />
      <LmcHandoverHistoryItem
        handover={
          Object {
            "logsByResident": Array [
              Object {
                "logs": Array [
                  Object {
                    "createdBy": "TestCarer",
                    "description": "This is a log",
                    "id": "TestId",
                  },
                ],
                "resident": Object {
                  "id": "TestId2",
                  "name": "TestResident",
                  "picture": "TestPicture",
                },
              },
            ],
            "notes": Array [
              Object {
                "createdBy": "TestCarer",
                "id": "TestId",
                "note": "This is a note",
              },
            ],
            "seenBy": Array [
              "TestId1",
              "TestId2",
              "TestId3",
            ],
          }
        }
      />
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div>
        <h2
          style={
            Object {
              "fontWeight": 300,
              "hyphens": "auto",
              "marginBottom": "0.3em",
              "marginTop": 20,
              "textOverflow": "ellipsis",
            }
          }
        >
          Mar 11th
        </h2>
        <div
          className="lmc-theme-gradient"
          style={
            Object {
              "height": 2,
              "marginBottom": 12,
              "width": "100%",
            }
          }
        />
        <LmcHandoverHistoryItem
          handover={
            Object {
              "logsByResident": Array [
                Object {
                  "logs": Array [
                    Object {
                      "createdBy": "TestCarer",
                      "description": "This is a log",
                      "id": "TestId",
                    },
                  ],
                  "resident": Object {
                    "id": "TestId2",
                    "name": "TestResident",
                    "picture": "TestPicture",
                  },
                },
              ],
              "notes": Array [
                Object {
                  "createdBy": "TestCarer",
                  "id": "TestId",
                  "note": "This is a note",
                },
              ],
              "seenBy": Array [
                "TestId1",
                "TestId2",
                "TestId3",
              ],
            }
          }
        />
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcHandoversHistory
        handovers={
          Object {
            "Mar 11th": Array [
              Object {
                "logsByResident": Array [
                  Object {
                    "logs": Array [
                      Object {
                        "createdBy": "TestCarer",
                        "description": "This is a log",
                        "id": "TestId",
                      },
                    ],
                    "resident": Object {
                      "id": "TestId2",
                      "name": "TestResident",
                      "picture": "TestPicture",
                    },
                  },
                ],
                "notes": Array [
                  Object {
                    "createdBy": "TestCarer",
                    "id": "TestId",
                    "note": "This is a note",
                  },
                ],
                "seenBy": Array [
                  "TestId1",
                  "TestId2",
                  "TestId3",
                ],
              },
            ],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcHandoversHistory {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handovers": Object {
            "Mar 11th": Array [
              Object {
                "logsByResident": Array [
                  Object {
                    "logs": Array [
                      Object {
                        "createdBy": "TestCarer",
                        "description": "This is a log",
                        "id": "TestId",
                      },
                    ],
                    "resident": Object {
                      "id": "TestId2",
                      "name": "TestResident",
                      "picture": "TestPicture",
                    },
                  },
                ],
                "notes": Array [
                  Object {
                    "createdBy": "TestCarer",
                    "id": "TestId",
                    "note": "This is a note",
                  },
                ],
                "seenBy": Array [
                  "TestId1",
                  "TestId2",
                  "TestId3",
                ],
              },
            ],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div>
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "marginTop": 20,
                  "textOverflow": "ellipsis",
                }
              }
            >
              Mar 11th
            </h2>
            <div
              className="lmc-theme-gradient"
              style={
                Object {
                  "height": 2,
                  "marginBottom": 12,
                  "width": "100%",
                }
              }
            />
            <LmcHandoverHistoryItem
              handover={
                Object {
                  "logsByResident": Array [
                    Object {
                      "logs": Array [
                        Object {
                          "createdBy": "TestCarer",
                          "description": "This is a log",
                          "id": "TestId",
                        },
                      ],
                      "resident": Object {
                        "id": "TestId2",
                        "name": "TestResident",
                        "picture": "TestPicture",
                      },
                    },
                  ],
                  "notes": Array [
                    Object {
                      "createdBy": "TestCarer",
                      "id": "TestId",
                      "note": "This is a note",
                    },
                  ],
                  "seenBy": Array [
                    "TestId1",
                    "TestId2",
                    "TestId3",
                  ],
                }
              }
            />
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div>
            <h2
              style={
                Object {
                  "fontWeight": 300,
                  "hyphens": "auto",
                  "marginBottom": "0.3em",
                  "marginTop": 20,
                  "textOverflow": "ellipsis",
                }
              }
            >
              Mar 11th
            </h2>
            <div
              className="lmc-theme-gradient"
              style={
                Object {
                  "height": 2,
                  "marginBottom": 12,
                  "width": "100%",
                }
              }
            />
            <LmcHandoverHistoryItem
              handover={
                Object {
                  "logsByResident": Array [
                    Object {
                      "logs": Array [
                        Object {
                          "createdBy": "TestCarer",
                          "description": "This is a log",
                          "id": "TestId",
                        },
                      ],
                      "resident": Object {
                        "id": "TestId2",
                        "name": "TestResident",
                        "picture": "TestPicture",
                      },
                    },
                  ],
                  "notes": Array [
                    Object {
                      "createdBy": "TestCarer",
                      "id": "TestId",
                      "note": "This is a note",
                    },
                  ],
                  "seenBy": Array [
                    "TestId1",
                    "TestId2",
                    "TestId3",
                  ],
                }
              }
            />
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcHandoversHistory
    handovers={
      Object {
        "Mar 11th": Array [
          Object {
            "logsByResident": Array [
              Object {
                "logs": Array [
                  Object {
                    "createdBy": "TestCarer",
                    "description": "This is a log",
                    "id": "TestId",
                  },
                ],
                "resident": Object {
                  "id": "TestId2",
                  "name": "TestResident",
                  "picture": "TestPicture",
                },
              },
            ],
            "notes": Array [
              Object {
                "createdBy": "TestCarer",
                "id": "TestId",
                "note": "This is a note",
              },
            ],
            "seenBy": Array [
              "TestId1",
              "TestId2",
              "TestId3",
            ],
          },
        ],
      }
    }
  />,
}
`;
